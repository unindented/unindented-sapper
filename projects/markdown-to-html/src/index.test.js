const markdown2html = require(".");

describe("markdown-to-html", () => {
  const text = `
Header 1
========

Header 2
--------

The quick brown fox jumped over the lazy
dog's back.

### Header 3

> This is a blockquote.
> 
> This is the second paragraph in the blockquote.
`;

  beforeEach(() => {
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  it("converts markdown to an html rehype tree", async () => {
    const result = await markdown2html("blog/example", text);
    return expect(result).toMatchSnapshot();
  });
});
