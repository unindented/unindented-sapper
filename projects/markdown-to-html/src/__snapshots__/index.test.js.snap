// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-to-html converts markdown to an html rehype tree 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Header 1",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "#",
            },
          ],
          "properties": Object {
            "class": "hash",
            "href": "#header-1",
          },
          "tagName": "a",
          "type": "element",
        },
      ],
      "properties": Object {
        "id": "header-1",
      },
      "tagName": "h1",
      "type": "element",
    },
    Object {
      "type": "text",
      "value": "
",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Header 2",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "#",
            },
          ],
          "properties": Object {
            "class": "hash",
            "href": "#header-2",
          },
          "tagName": "a",
          "type": "element",
        },
      ],
      "properties": Object {
        "id": "header-2",
      },
      "tagName": "h2",
      "type": "element",
    },
    Object {
      "type": "text",
      "value": "
",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "The quick brown fox jumped over the lazy
dogâ€™s back.",
        },
      ],
      "properties": Object {},
      "tagName": "p",
      "type": "element",
    },
    Object {
      "type": "text",
      "value": "
",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Header 3",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "#",
            },
          ],
          "properties": Object {
            "class": "hash",
            "href": "#header-3",
          },
          "tagName": "a",
          "type": "element",
        },
      ],
      "properties": Object {
        "id": "header-3",
      },
      "tagName": "h3",
      "type": "element",
    },
    Object {
      "type": "text",
      "value": "
",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "
",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "This is a blockquote.",
            },
          ],
          "properties": Object {},
          "tagName": "p",
          "type": "element",
        },
        Object {
          "type": "text",
          "value": "
",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "This is the second paragraph in the blockquote.",
            },
          ],
          "properties": Object {},
          "tagName": "p",
          "type": "element",
        },
        Object {
          "type": "text",
          "value": "
",
        },
      ],
      "properties": Object {},
      "tagName": "blockquote",
      "type": "element",
    },
  ],
  "properties": Object {
    "class": "rehype",
  },
  "tagName": "div",
  "type": "element",
}
`;
